### PR TITLE
[13.0][IMP]bi_sql_editor: archive Scheduled Action to not recreate when resetting the SQL View

### PR DESCRIPTION
**Pull request by GuillemCForgeFlow**
Based on the agreed solution: https://github.com/OCA/reporting-engine/pull/903#pullrequestreview-2273835493. It is best to archive the Scheduled Action every time you set back to the draft state. Then, when creating the SQL view, if it is a materialized view, unarchive the Scheduled Action if already exists, otherwise create it from scratch.

Then, if the SQL View is deleted, also make sure to remove the Scheduled Action if there is one assigned as it has not been done in the `button_set_draft` method

cc @ForgeFlow

### CODE REVIEW

File: bi_sql_editor/models/bi_sql_view.py [hunk 1]
```diff
@@ -324,7 +329,7 @@ def button_set_draft(self):
             sql_view.pivot_view_id.unlink()
             sql_view.search_view_id.unlink()
             if sql_view.cron_id:
-                sql_view.cron_id.unlink()
```
Comment (legalsylvain) [inline bi_sql_editor/models/bi_sql_view.py:327]: As you remove unlink here, you should add if the bi.sql.view is deleted I think.

Apart that point, LGTM. Very ligther solution, and nice improvment ! thanks !

Comment (GuillemCForgeFlow) [reply bi_sql_editor/models/bi_sql_view.py:327]: sure, thank you for noticing 👍🏿 just applied the changes, we should be good now 😄

**Review by legalsylvain — APPROVED** [review]
Thanks !

File: bi_sql_editor/models/bi_sql_view.py [hunk 2]
```diff
@@ -281,6 +282,9 @@ def unlink(self):
                     "If you want to delete them, first set them to draft."
                 )
             )
+        for sql_view in self:
+            if sql_view.cron_id:
+                sql_view.cron_id.unlink()
```
Comment (legalsylvain) [inline bi_sql_editor/models/bi_sql_view.py]: Nitpicking : The 3 lines could be replaced by seld.mapped("cron_id").unlink() i think.

Comment (GuillemCForgeFlow) [reply bi_sql_editor/models/bi_sql_view.py]: no problem, I also think it looks cleaner like this 👍🏿

**Comment by legalsylvain** [comment]
/ocabot merge minor

Thanks !

**Comment by legalsylvain** [comment]
@GuillemCForgeFlow could you port this patch upstream ? (17.0 and ideally 16.0)

thanks !

**Comment by GuillemCForgeFlow** [comment]
> @GuillemCForgeFlow could you port this patch upstream ? (17.0 and ideally 16.0)
> 
> thanks !

sure, I will try to allocate some time during the next few days, will keep you posted 👍🏿

**Comment by GuillemCForgeFlow** [comment]
Hi @legalsylvain, I've seen that you already added some similar changes back in the day from v16 on: https://github.com/OCA/reporting-engine/commit/680bfdfdada245716307e49687312717960d2f88

So I guess we should just leave this change in v13 😄

**Comment by legalsylvain** [comment]
> Hi @legalsylvain, I've seen that you already added some similar changes back in the day from v16 on: https://github.com/OCA/reporting-engine/commit/680bfdfdada245716307e49687312717960d2f88

ahah ! I didn't remember.

**Pull request merged**